packages/paste-website/src/pages/patterns/data-export/index.mdx

```diff
@@ -0,0 +1,295 @@
+  will be exported. E.g. “Export CSV”.
+- Tell the customer if there are any limitations to the data that can be exported.
+  Use a [Tooltip](/components/tooltip) to communicate this information.
+
+<LivePreview scope={{Tooltip, Button, ExportIcon}} noInline language="jsx">
+  {exportButton}
+</LivePreview>
+
+Using the word “Export” and the ExportIcon are predictable signals to customers that they can extract data from Twilio.
+
+Twilio products generally enable customers to convert files to a range of formats before downloading and saving. However if you can’t provide a format conversion, you should still use the word “Export” and the ExportIcon.
+
+The [delayed export](#delayed-export) is an exception to this copy guideline and uses the word “Download” and the DownloadIcon instead. action is placed at the top right of the area where the object will appear once it has been created. In this example, the entire page is dedicated to displaying the user’s services, so the “create new service” action is placed at the top of the page next to the page header.
+
+## Variations
+
+### Immediate export
+
+An immediate export happens when a customer can download their export on
+the same page where they clicked the export Button or Menu item.
+
+When using a Button, while an export is in progress, show the Button loading
+state. When it’s in a loading state, it will disable by default.
+
+<LivePreview
+  scope={{Stack, MenuButton, MenuItem, Menu, Tooltip, useMenuState, Button, ChevronDownIcon, ExportIcon}}
+  noInline
+  language="jsx"
+>
+  {exportImmediate}
+</LivePreview>
+
+Most often, selecting the export Button or Menu item should trigger a “save as”
+dialog in the browser. Customers will be able to track the export’s download
+status through their browser.
+
+However in cases where you might offer more customization of the data, you
+can trigger a Modal first. In the Modal, customers can select what columns
+to export, additional file formats, etc.
+
+<LivePreview
+  scope={{
+    useUIDSeed,
+    CheckboxGroup,
+    Checkbox,
+    Modal,
+    ModalHeader,
+    ModalHeading,
+    ModalBody,
+    ModalFooter,
+    ModalFooterActions,
+    Tooltip,
+    Button,
+    ExportIcon,
+  }}
+  noInline
+  language="jsx"
+>
+  {exportModal}
+</LivePreview>
+
+### Delayed export
+
+A delayed export happens when a customer needs to wait for data to be ready.
+Use the same components as the immediate export pattern, but make sure to:
+
+1. Confirm the export was initiated by using a Toast component. Inform the customer that they’ll receive an email notification when it’s ready.
+2. Send an email notification to the customer when the export is ready, with a link to or attachment of the file(s). In this second step, use the word “Download” (example: “Download CSV”) to trigger the actual file download since the customer already clicked an export action to start the export process.
```

Comment:
It might make more sense to separate these steps out so they appear above 2 separate code examples.

I'll tag you in an edit on the spec.